src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs

```diff
@@ -148,6 +150,84 @@ public WorkspaceInfo()
             }
         }
 
+        internal static bool FromJsonDocument(String jsonDoc, string path, bool isTestMode,
+            bool forceManualExecutionMode, ILogger logger, out WorkspaceInfo workspaceInfo)
+        {
+            var jObject = (JObject)JsonConvert.DeserializeObject(jsonDoc);
+            try
+            {
+                string funName = null;
+                double cx = 0;
+                double cy = 0;
+                double zoom = 1.0;
+                double scaleFactor = 1.0;
+                string id = "";
+                string category = "";
+                string description = "";
+                string version = "";
+                var runType = RunType.Manual;
+                int runPeriod = RunSettings.DefaultRunPeriod;
+                bool hasRunWithoutCrash = false;
+                bool isVisibleInDynamoLibrary = true;
+
+                cx = double.Parse(jObject["View"]["X"].ToString(), CultureInfo.InvariantCulture);
+                cy = double.Parse(jObject["View"]["Y"].ToString(), CultureInfo.InvariantCulture);
+                zoom = double.Parse(jObject["View"]["Zoom"].ToString(), CultureInfo.InvariantCulture);
+                scaleFactor = double.Parse(jObject["View"]["Dynamo"]["ScaleFactor"].ToString(), CultureInfo.InvariantCulture);
+                funName = jObject["Name"].ToString();
+                id = jObject["Uuid"].ToString();
+                category = jObject["Category"].ToString();
+                description = jObject["Description"].ToString();
+                hasRunWithoutCrash = bool.Parse(jObject["View"]["Dynamo"]["HasRunWithoutCrash"].ToString());
+                isVisibleInDynamoLibrary = bool.Parse(jObject["View"]["Dynamo"]["IsVisibleInDynamoLibrary"].ToString());
+                version = jObject["View"]["Dynamo"]["Version"].ToString();
+                if (forceManualExecutionMode || !Enum.TryParse(jObject["View"]["Dynamo"]["RunType"].ToString(), false, out runType))
+                {
+                    runType = RunType.Manual;
+                }
+                runPeriod = Int32.Parse(jObject["View"]["Dynamo"]["RunPeriod"].ToString());
+
+                // we have a dyf and it lacks an ID field, we need to assign it
+                // a deterministic guid based on its name.  By doing it deterministically,
+                // files remain compatible
+                if (string.IsNullOrEmpty(id) && !string.IsNullOrEmpty(funName) && funName != "Home")
```

Comment:
I feel like this check is very brittle with the "Home" check.

So two things:
1. is there some other way to accomplish this check?
2. can you pull out all magic strings for property names in this function and maybe start pulling them out in other serialization contexts as well and use constants defined somewhere so if we do change them it's quite clear where they need to be changed. 

I think this might as well be a good as any place to start this refactor, though I don't want to hold up this PR for the other stuff, so your call how far you think it is worth doing (or at all)

Reply:
1. I can check for field "IsCustomNode" but that seems will disappear soon as well. Do you have any suggestion? Right now I just followed XML file logic.
2. Will do for `Home`